flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/cluster/JobManagerLogListHandlerTest.java

```diff
@@ -0,0 +1,142 @@
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		mockRestfulGateway = new TestingDispatcherGateway.Builder().build();
+	}
+
+	@Test
+	public void testGetJobManagerLogsList() throws Exception {
+		String jobmanagerLogContent = "jobmanager log content";
+		String jobmanagerStdoutContent = "jobmanager stdout content";
+		String jobmanagerCustomLogContent = "jobmanager custom log content";
+
+		LogInfo jobmanagerLog = new LogInfo("jobmanager.log", jobmanagerLogContent.length());
+		LogInfo jobmanagerStdout = new LogInfo("jobmanager.out", jobmanagerStdoutContent.length());
+		LogInfo customLog = new LogInfo("test.log", jobmanagerCustomLogContent.length());
+
+		Map<LogInfo, String> logInfo2ContentMap = new HashMap<>(3);
+		logInfo2ContentMap.put(jobmanagerLog, jobmanagerLogContent);
+		logInfo2ContentMap.put(jobmanagerStdout, jobmanagerStdoutContent);
+		logInfo2ContentMap.put(customLog, jobmanagerCustomLogContent);
+
+		Map<String, LogInfo> fileNameAndLogInfoMap = new HashMap<>(3);
+		fileNameAndLogInfoMap.put(jobmanagerLog.getName(), jobmanagerLog);
+		fileNameAndLogInfoMap.put(jobmanagerStdout.getName(), jobmanagerStdout);
+		fileNameAndLogInfoMap.put(customLog.getName(), customLog);
+
+		JobManagerLogListHandler jobManagerLogListHandler = createHandler(logInfo2ContentMap);
+		LogListInfo logListInfo = jobManagerLogListHandler.handleRequest(testRequest, mockRestfulGateway).get();
+
+		assertThat(logListInfo.getLogInfos(), hasSize(logInfo2ContentMap.size()));
+		for (LogInfo logInfo : logListInfo.getLogInfos()) {
+			assertEquals(logInfo, fileNameAndLogInfoMap.get(logInfo.getName()));
+		}
+	}
+
+	@Test
+	public void testGetJobManagerLogsListWhenLogDirIsNull() throws Exception {
+		JobManagerLogListHandler jobManagerLogListHandler = createHandler(Collections.emptyMap());
+		LogListInfo logListInfo = jobManagerLogListHandler.handleRequest(testRequest, mockRestfulGateway).get();
+		assertThat(logListInfo.getLogInfos(), is(empty()));
+	}
+
+	private JobManagerLogListHandler createHandler(Map<LogInfo, String> logInfo2ContentMap) {
+		WebMonitorUtils.LogFileLocation logFileLocation = createLogFileLocation(logInfo2ContentMap);
+		return new JobManagerLogListHandler(
+			() -> CompletableFuture.completedFuture(null),
+			TestingUtils.TIMEOUT(),
+			Collections.emptyMap(),
+			JobManagerLogListHeaders.getInstance(),
+			logFileLocation.logDir);
+	}
+
+	private WebMonitorUtils.LogFileLocation createLogFileLocation(Map<LogInfo, String> logInfo2ContentMap) {
+		Configuration config = new Configuration();
+		try {
+			for (Map.Entry<LogInfo, String> logInfo2Content : logInfo2ContentMap.entrySet()) {
+				LogInfo logInfo = logInfo2Content.getKey();
+				String content = logInfo2Content.getValue();
+				File file = temporaryFolder.newFile(logInfo.getName());
+				FileUtils.writeStringToFile(file, content);
```

Comment:
We should make sure that the file is UTF-8 or ASCII encoded otherwise we risk that one ASCII character is not one byte. I fixed it before merging.

Reply:
I agree with you. 
I find the same code in WebFrontendITCase, whether need we to fix them?